alternator/executor.cc

```diff
@@ -2665,8 +2689,44 @@ update_item_operation::apply(std::unique_ptr<rjson::value> previous_item, api::t
                 const rjson::value& value = (it->value)["Value"];
                 validate_value(value, "AttributeUpdates");
                 do_update(std::move(column_name), value);
+            } else if (action == "ADD") {
```

Comment:
side note: with this routine, `update_item_operation::apply` hit >200 lines of code - maybe we should consider moving some of the subroutines out in a follow-up, to make it fit into a single screen

Reply:
Perhaps. My philosophy on splitting function is apparently very different from other developers' - whereas some view any function with more than 20 lines an abomination, I view a long linear function (which you can just read linearly to understand) better than artificially splitting it into 10 different functions that nobody will understand when reading them one by one. That doesn't say that we can't split it in a way that doesn't look artificial. I can try to think about this later.